src/content/effective-dart/design.md

```diff
@@ -1678,11 +1667,13 @@ bool convertToBool(Object arg) {
 }
 ```
 
-The main exception to this rule is when working with existing APIs that use
-`dynamic`, especially inside a generic type. For example, JSON objects have type
-`Map<String, dynamic>` and your code will need to accept that same type. Even
-so, when using a value from one of these APIs, it's often a good idea to cast it
-to a more precise type before accessing members.
+Prefer using `Object?` over `dynamic` in code that is not invoking a member
+dynamically, even when working with existing APIs that use `dynamic`.
+For example, JSON objects have runtime type `Map<String, dynamic>`, which is an
+equivalent type to `Map<String, Object?>`.
+When using a value from one of these APIs, it's often a good idea to treat it as
```

Comment:
Which APIs is this? (First use of "API" here, but the definite form suggests that it refers to something mentioned earlier.)

Is it something that expects a `Map<String, dynamic>`, or which produces it?
Could the previous paragraph start with
"APIs which accept or produce JSON-object maps traditionally type those as `Map<String, dynamic>` ..."